src/plugins/dashboard/public/application/listing/create_button.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React, { useState } from 'react';
+import {
+  EuiButton,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexItem,
+  EuiPopover,
+} from '@elastic/eui';
+import type { DashboardCreator, DashboardCreators } from '../../types';
+
+interface CreateButtonProps {
+  dashboardItemCreators?: () => DashboardCreators;
+}
+
+const CreateButton = (props: CreateButtonProps) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setPopover(false);
+  };
+
+  const getItems = (creators: DashboardCreators) => {
+    return creators.map((creator: DashboardCreator) => (
+      <EuiContextMenuItem key={creator.id} onClick={creator.creatorFn}>
+        {creator.defaultText}
+      </EuiContextMenuItem>
+    ));
+  };
+
+  const forceSingleButtonOnly = false;
+
+  const creators: DashboardCreators = props.dashboardItemCreators
+    ? props.dashboardItemCreators()
+    : [];
+
+  const renderMenu = () => {
+    if (creators.length === 0) {
+      return null;
+    } else if (forceSingleButtonOnly || creators.length === 1) {
+      const creator = creators[0];
+      return (
+        <EuiFlexItem grow={false}>
+          <EuiButton
+            onClick={creator.creatorFn}
+            data-test-subj="newItemButton"
+            iconType="plusInCircle"
+            fill
+          >
+            Create {creator.defaultText}
```

Comment:
```suggestion
            <FormattedMessage
              id="opensearch-dashboards-react.tableListView.listing.createNewItemButtonLabel"
              defaultMessage="Create {entityName}"
              values={{ entityName: this.props.entityName }}
            />
```

Reply:
Yes, I have recovered this JSX into the new CreateButton component